detail.py

```diff
@@ -0,0 +1,25 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from catalog import models as cmod
+
+@view_function
+def process_request(request, prod:cmod.Product):
+
+    for index, item in enumerate(request.last_five):
```

Comment:
Using enumerate is very clever; however, I think you can just loop through request.last_five like:
for item in request.last_five: